tools/shared/src/main/scala/org/scalajs/core/tools/javascript/JSDesugaring.scala

```diff
@@ -1576,8 +1576,24 @@ private[javascript] object JSDesugaring {
           }
 
         case ApplyStatically(receiver, cls, method, args) =>
-          val fun = encodeClassVar(cls.className).prototype DOT method
-          js.Apply(fun DOT "call", (receiver :: args) map transformExpr)
+          val className = cls.className
+          val transformedArgs = (receiver :: args) map transformExpr
+
+          if (classEmitter.isInterface(className)) {
+            val Ident(methodName, origName) = method
+            if (isStrongMode) {
+              js.Apply(js.DotSelect(envField("c", className),
+                  js.Ident("$f_" + methodName, origName)(method.pos)),
+                  transformedArgs)
```

Comment:
In strong mode, we need top-level functions to be static methods of a class, because only consecutive classes can see each other. Whether they are static methods or default methods at the IR level, they should become top-level functions, but in strong mode they become ES static methods.

There's no collision because no IR method can start with `$f_`.